docs/proposals/004_carbon_brd.md

```diff
@@ -0,0 +1,109 @@
+
+## Introduction
+
+The purpose of this document is to outline the requirements for hosting and loading extensions within the **Carbon** architecture. These extensions can take various forms, such as standalone runnable apps, embedded iframes, or remotely hosted apps (e.g., served via a CDN). This document details the core features related to routing, loading mechanisms, communication protocols, and security measures for enabling extensibility within the **Carbon** platform.
+
+## Business Objectives
+
+1. **Extendability:** The **Carbon** system should enable seamless integration of third-party apps (extensions) into the platform. These extensions can be colocated with the host platform, remotely hosted, or standalone apps. The system must handle various app types, including micro frontends, iframe-based apps, or static standalone applications, by providing standardized integration mechanisms.
+2. **Scalability:** The **Carbon** platform must handle the loading and management of multiple extensions simultaneously, without negatively impacting performance, security, or maintainability. Whether these extensions are colocated or hosted remotely, the system should handle an increasing number of apps without compromising the overall stability of the platform.
+
+3. **Flexibility:** The **Carbon** system should allow extensions to be configurable via a manifest or configuration file, allowing for a unified approach to manage loading, dependencies, and initialization. This configuration-driven approach ensures that local and remote extensions are managed flexibly while enabling dynamic routing, props, and parameters to be set during initialization.
+
+4. **Isolation:** Extensions must operate in full isolation from the host platform and from other extensions. This ensures that each extension is sandboxed, preventing any interference with the host's state, routing, or security model. For iframe-based and remote extensions, communication should be securely managed via defined APIs, preserving the integrity of the **Carbon** platform.
+
+## Scope
+
+This document covers the following aspects of extension integration within **Carbon**:
+
+- Routing for extensions
+- Loading mechanisms for colocated and remote apps
+- Communication between the host app and extensions
+- Security considerations for external extensions
+
+---
+
+## Key Features
+
+### 1. Routing
+
+Routing is essential for isolating and providing direct access to each extension within the **Carbon** architecture.
+
+#### Requirements:
+
+- **Dedicated Route per App:** Each extension should have its own unique route, such as `/app/:appId`, which should support both local and remote navigation.
+- **Isolated Routing:** Extensions should not modify or interfere with the base URL of the host app. Any internal routing within an extension must remain scoped to that extension, ensuring that changes in URL or navigation do not affect the host's routing logic.
+
```

Comment:
```suggestion
- **Cross Linking Support:** While extensions should not be able to modify the base URL directly the host app should provide an interface with which extension can cross link to each other
```

Reply:
> We probably want something like this, too.

We decided to include these details in the Interface section, but I believe it's worth mentioning them here as well.